proto/proto/block_producer.proto

```diff
@@ -2,22 +2,18 @@
 syntax = "proto3";
 package block_producer;
 
-import "requests.proto";
-import "responses.proto";
-
-import "block.proto";
-import "digest.proto";
-import "note.proto";
-import "transaction.proto";
-
+import "types/account.proto";
+import "types/blockchain.proto";
+import "types/primitives.proto";
+import "types/shared.proto";
 import "google/protobuf/empty.proto";
 
 service Api {
     // Submits proven transaction to the Miden network
-    rpc SubmitProvenTransaction(requests.SubmitProvenTransactionRequest) returns (responses.SubmitProvenTransactionResponse) {}
+    rpc SubmitProvenTransaction(shared.SubmitProvenTransaction) returns (shared.ProvenTransaction) {}
```

Comment:
A couple of comments here:

- I don't think `ProvenTransaction` is the right message name here. What we are returning is just the block height and calling it `ProvenTransaction` is going to be confusing. I'm open to other suggestions, but I'd strongly prefer `SubmitProvenTransactionResponse` over `ProvenTransaction` here.
  - Also, this message should probably be defined here and then imported in other files from here as needed.
- `SubmitProvenTransaction` could actually be called `ProvenTransaction` because it in fact contains a proven transaction. I would put this message into `types/transaction.proto` together with other transaction-related messages (e.g., `TransactionId`).